### PR TITLE
added some regex based checks for dangerous commands

### PR DESCRIPTION
**Pull request by Kvadratni**
This pull request enhances the safety of shell command execution by introducing a check for dangerous commands and updating the notification messages accordingly. The most important changes include adding a utility function to identify dangerous commands, modifying the shell command execution logic to incorporate this check, and adding tests to validate the new functionality.

### Safety Enhancements:

* [`src/goose/utils/check_shell_command.py`](diffhunk://#diff-49bef33f51eea6dab40ca972595545c116f882e07a153c0e32bbb456d881535cR1-R33): Added a new utility function `is_dangerous_command` to identify potentially dangerous shell commands based on predefined patterns.
* [`src/goose/toolkit/developer.py`](diffhunk://#diff-3e5431e1f0b4918b17f7788a405d3b3e22610eb8f91043afb1a6042e41c039a6L159-R166): Updated the `shell` method to use the `is_dangerous_command` function to check for dangerous commands before execution.

### Notification Improvements:

* [`src/goose/toolkit/developer.py`](diffhunk://#diff-3e5431e1f0b4918b17f7788a405d3b3e22610eb8f91043afb1a6042e41c039a6L138-R139): Changed the status message from "running shell command" to "planning to run shell command" before the danger check, and moved the "running shell command" message to after the check. [[1]](diffhunk://#diff-3e5431e1f0b4918b17f7788a405d3b3e22610eb8f91043afb1a6042e41c039a6L138-R139) [[2]](diffhunk://#diff-3e5431e1f0b4918b17f7788a405d3b3e22610eb8f91043afb1a6042e41c039a6L159-R166)

### Testing:

* [`tests/utils/test_check_shell_command.py`](diffhunk://#diff-ebfea0d86f26cad7e69859942f9bfb28274e30ec08a6badcafc9e245b36f2d68R1-R27): Added tests to verify that the `is_dangerous_command` function correctly identifies dangerous commands and does not flag safe commands.

### Import Adjustments:

* [`src/goose/toolkit/developer.py`](diffhunk://#diff-3e5431e1f0b4918b17f7788a405d3b3e22610eb8f91043afb1a6042e41c039a6R4): Added an import statement for the `is_dangerous_command` function.

### CODE REVIEW

File: src/goose/utils/check_shell_command.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    Check if the command matches any dangerous patterns.
+
+    Args:
+        command (str): The shell command to check.
+
+    Returns:
+        bool: True if the command is dangerous, False otherwise.
+    """
+    dangerous_patterns = [
+        r"\brm\b",  # rm command
```
Comment (baxen) [inline src/goose/utils/check_shell_command.py:15]: nit: I think `rm` on a single file might be fine, but I am scared of `rm -r` - especially combined with the file path checks below. Wdyt?

File: tests/utils/test_check_shell_command.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        "chown root:root /etc/passwd",
+        "mkfs -t ext4 /dev/sda1",
+        "systemctl stop nginx",
+        "reboot",
+        "shutdown now",
+        "echo hello > ~/.bashrc",
+    ],
+)
+def test_dangerous_commands(command):
+    assert is_dangerous_command(command)
```
Comment (baxen) [inline tests/utils/test_check_shell_command.py:22]: nit: probably want to assert some of the inverse, with safe commands too

**Review by baxen — APPROVED** [review]
LGTM! Left a couple of small detail comments

File: tests/utils/test_check_shell_command.py (hunk 2 above)
Comment (Kvadratni) [reply tests/utils/test_check_shell_command.py:22]: that's what the second test does.

File: src/goose/utils/check_shell_command.py (hunk 1 above)
Comment (Kvadratni) [reply src/goose/utils/check_shell_command.py:15]: better safe than sorry.

Comment (scizzorz) [reply src/goose/utils/check_shell_command.py:15]: A bit late, but I think it'd be nice to add some user flexibility here, possibly via config setting somewhere. eg: one might also not want to allow `curl`, `http`, etc.

**Pull request merged**